llvm/tools/llvm-exegesis/lib/ValidationEvent.cpp

```diff
@@ -0,0 +1,53 @@
+
+#include "ValidationEvent.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/Errc.h"
+#include "llvm/Support/Error.h"
+
+namespace llvm {
+namespace exegesis {
+
+namespace {
+
+struct ValidationEventInfo {
+  const char *const Name;
+  const char *const Description;
+};
+
+// Information about validation events, indexed by `ValidationEvent` enum
+// value.
+static constexpr ValidationEventInfo ValidationEventInfos[NumValidationEvents] =
```

Comment:
Would it be possible to use a `static_assert` to ensure that the length of this array is the same as the enum to prevent someone missing the implementation of a validation counter here? I think it would be tricky with a standard array, but with a `std::array`, I think we should have access to a `constexpr` `size()` method.

Reply:
That was my intent by specifying the array bound. I had forgotten that `int arr[3] = {1,2};` compiled perfectly fine in c++ :( Done, thanks.